src/WebJobs.Script/Diagnostics/Extensions/LoggerExtension.cs

```diff
@@ -183,6 +183,21 @@ internal static class LoggerExtension
             new EventId(337, nameof(MinimumBundleVersionNotSatisfied)),
             "Referenced bundle {bundleId} of version {bundleVersion} does not meet the required minimum version of {minimumVersion}. Update your extension bundle reference in host.json to reference {minimumVersion2} or later.");
 
+        private static readonly Action<ILogger, string, Exception> _validateHostJsonZipIssue =
+    LoggerMessage.Define<string>(LogLevel.Information,
+        new EventId(338, nameof(ValidateHostJsonZipIssue)),
+        "No functions were found. The host.json file was not located at the root, but it was found at the following path(s): {hostJsonFilesPath}. A default host.json was created at the root using a default configuration to start the function app. See https://aka.ms/deployment-zip-push for more information on deployment.");
+
+        private static readonly Action<ILogger, Exception> _validateHostJsonNoHostJson =
+            LoggerMessage.Define(LogLevel.Information,
+                new EventId(339, nameof(ValidateHostJsonNoHostJson)),
+                "No functions were found. The host.json file was not located at the root. A default host.json was created at the root using a default configuration to start the function app. See https://aka.ms/functions-deployment-technologies for more information on deployment.");
```

Comment:
Is this the case where no `host.json` file is found at all? If so, can this be expanded to "host.json file was not found in the deployment payload" or something similar, instead of "not located at the root"?